File: Net/Patterns/TemplateMethod/Before/Coffee.cs
namespace Patterns.TemplateMethod.Before
{
    using System;

    public class Coffee
    {
        public void PrepareRecipe()
        {
            BoilWater();
            BrewCoffeeGrinds();
            PourInCup();
            AddSugarAndMilk();
        }

        private void BoilWater()
        {
            Console.WriteLine(BeverageMessages.BoilWater);
        }

        private void BrewCoffeeGrinds()
        {
            Console.WriteLine(BeverageMessages.BrewCoffeeGrinds);
        }

        private void PourInCup()
        {
            Console.WriteLine(BeverageMessages.PourInCup);
        }

        private void AddSugarAndMilk()
        {
            Console.WriteLine(BeverageMessages.AddSugarAndMilk);
        }
    }
}

File: Java/TemplateMethod/Before/src/main/templatemethod/Coffee.java
package templatemethod;

public class Coffee {
 
	public void prepareRecipe() {
		boilWater();
		brewCoffeeGrinds();
		pourInCup();
		addSugarAndMilk();
	}
 
	private void boilWater() {
		System.out.println(BeverageMessages.BoilWater);
	}
 
	private void brewCoffeeGrinds() {
		System.out.println(BeverageMessages.BrewCoffeeGrinds);
	}
 
	private void pourInCup() {
		System.out.println(BeverageMessages.PourInCup);
	}
 
	private void addSugarAndMilk() {
		System.out.println(BeverageMessages.AddSugarAndMilk);
	}
}
